src/endpoints/identifier/services/add.ts

```diff
@@ -35,7 +35,12 @@ export function add (request: Request): Response {
 
   // Get the service JSON from the request and check that
   // is correctly formed.
-  const service = <Service>request.body.json();
+  let service : Service;
+  if (authenticatedIdentity.policy === 'user_cose_sign1') {

```

Comment:
This could be encapsulated into an "authenticated call" object and re-used in the other endpoints.

Reply:
I've tried to do this sort of thing, but because some requests use Json and others strings I've added some methods to a authenticated request for this purpose